platform/features/table/src/controllers/MCTTableController.js

```diff
@@ -51,22 +61,84 @@ define(
                 self.updateRows($scope.rows);
             };
 
+            /*
+             * Define watches to listen for changes to headers and rows.
+             */
             $scope.$watchCollection('filters', function () {
-                self.updateRows(self.$scope.rows);
+                self.updateRows($scope.rows);
             });
-            $scope.$watchCollection('headers', this.updateHeaders.bind(this));
-            $scope.$watchCollection('rows', this.updateRows.bind(this));
+            $scope.$watch('headers', this.updateHeaders.bind(this));
+            $scope.$watch('rows', this.updateRows.bind(this));
+
+            /*
+             * Listen for rows added individually (eg. for real-time tables)
+             */
+            $scope.$on('addRow', this.newRow.bind(this));
         }
 
         /**
-         * On scroll, calculate which rows indexes are visible and
-         * ensure that an equal number of rows are preloaded for
-         * scrolling in either direction.
+         * If auto-scroll is enabled, this function will scroll to the
+         * bottom of the page
+         * @private
+         */
+        MCTTableController.prototype.scrollToBottom = function() {
+            var self = this;
+
+            //Use timeout to defer execution until next digest when any
+            // pending UI changes have completed, eg. a new row in the table.
+            if (this.$scope.autoScroll) {
+                this.$timeout(function(){
+                        self.scrollable[0].scrollTop = self.scrollable[0].scrollHeight;
+                });
+            }
+        };
+
+        /**
+         * Handles a row add event. Rows can be added as needed using the
+         * `addRow` broadcast event.
+         * @private
+         */
+        MCTTableController.prototype.newRow = function (event, row) {
+            //Add row to the filtered, sorted list of all rows
+            if (this.filterRows([row]).length > 0) {
+                this.insertSorted(this.$scope.displayRows, row);
+            }
+
+            //Keep 'rows' synchronized as it provides the unsorted,
+            // unfiltered model for this view
+            if (!this.$scope.rows) {
+                this.$scope.rows = [];
+            }
+            this.$scope.rows.push(row);
```

Comment:
Yeah, that's a fair point. I actually originally implemented it with the parent being responsible for updating `rows`. In the interests of performance, I used the row passed with the broadcast anyway. Trying to do a delta on `rows` would be too costly, and there's no guarantee that the last row is the new one. 

Given that I was using the row object passed in by the broadcast, I would want to guarantee that rows was in sync and not leave it up to the developer in the calling code to know that they need to do it. So, given that, I ended up making MctTable responsible for keeping them in sync

Conceptually, if you were to map my implementation to a notional class-type interface, it would be
* `'$scope.rows = ...` -> `setRows(rows[])`
*  `$scope.broadcast(...)` -> `addRow(row)`

As a developer I would be confused if setting rows initially was automatically picked up by MctTable, but adding a row later required me to explicitly inform it.